packages/monaco/src/browser/monaco-quick-input-service.ts

```diff
@@ -182,8 +182,23 @@ export class MonacoQuickInputService implements QuickInputService {
     }
 
     input(options?: InputOptions, token?: CancellationToken): Promise<string | undefined> {
-        return this.monacoService.input();
+        let inputOptions: monaco.quickInput.IInputOptions | undefined = undefined;
+        if (options) {
+            inputOptions = {
+                ignoreFocusLost: options?.ignoreFocusLost,
+                password: options?.password,
+                placeHolder: options?.placeHolder,
+                prompt: options?.prompt,
+                value: options?.value,
+                valueSelection: options?.valueSelection
+            };
```

Comment:
Why not use the spread operator on the options here like `inputOptions = { ...options }`? Is there a benefit in manually assembling the options? Is there a special reason why we don't pass the options directly?

Reply:
@msujew I could not use the spread since the `validateInput` typings are different for `InputOptions` and `IInputOptions`, and we'd get a compilation error.

I could however do the following, or perhaps align the typings:

```ts
input(options?: InputOptions, token?: CancellationToken): Promise<string | undefined> {
    let inputOptions: monaco.quickInput.IInputOptions | undefined = undefined;
    if (options) {
        let validateInput = undefined;
        if (options.validateInput) {
            validateInput = (input: string) => options!.validateInput!(input)!;
        }
        inputOptions = {...options, validateInput};
    }
    return this.monacoService.input(inputOptions, token);
}
```

Reply:
Alright I see, I didn't notice the difference in typing previously. That's alright with me 👍